lib/CustomFields/components/FieldAccordion/edit-fields/shared-fields/OptionsField/OptionsField.js

```diff
@@ -145,26 +126,32 @@ const OptionsField = ({
                         <Field
                           component={Checkbox}
                           type="checkbox"
-                          name={`${fieldNamePrefix}.selectField.defaults`}
-                          value={fields.value[rowData.rowIndex]}
-                          checked={
-                            (fields.value[rowData.rowIndex] !== '') &&
-                            defaultOptions.includes(fields.value[rowData.rowIndex])
-                          }
-                          disabled={fields.value[rowData.rowIndex] === ''}
+                          name={`${fieldNamePrefix}.selectField.options.values[${rowData.rowIndex}].default`}
+                          checked={fields.value[rowData.rowIndex].default}
+                          disabled={fields.value[rowData.rowIndex].value === ''}
                         />
                       ) : (
                         <Field
                           component={RadioButton}
-                          name={`${fieldNamePrefix}.selectField.defaults`}
-                          value={rowData.rowIndex
-                            ? fields.value[rowData.rowIndex - 1]
-                            : NO_DEFAULT_OPTIONS_VALUE
-                          }
+                          name={`${fieldNamePrefix}.selectField.options.values[${rowData.rowIndex - 1}].default`}
                           marginBottom0
                           type="radio"
                           className={css.defaultControl}
-                          disabled={fields.value[rowData.rowIndex - 1] === ''}
+                          value={rowData.rowIndex
+                            ? fields.value[rowData.rowIndex - 1].id
+                            : NO_DEFAULT_OPTIONS_VALUE
+                          }
+                          checked={(!rowData.rowIndex && noDefaultOptionsSelected)
+                            || fields.value[rowData.rowIndex - 1]?.default === true
```

Comment:
We need checking for `true` here because checkbox in the first row (that doesn't have an option for it) has value `NO_DEFAULT_OPTIONS_VALUE`.
In this case we need to check for a boolean value explicitly